test/Interop/Cxx/members/Inputs/constructor-calls-function-from-nested-struct.h

```diff
@@ -0,0 +1,17 @@
+#ifndef TEST_INTEROP_CXX_MEMBERS_INPUTS_CONSTRUCTOR_CALLS_FUNCTION_FROM_NESTED_STRUCT_H
```

Comment:
Does `Cxx/members` already exist? If not, can we name it `member` (no "s") so that it's more in line with the other test directories?

Also, any of the `callConstructor` tests should go in the `/class` directory. And arguable so should all the tests.

Reply:
Also, is there any way we combine some of these tests? I don't see any reason we need half-a-dozen new headers and a dozen new test files. It will make things more difficult to find in the future. 

Maybe you could add some of these to the existing member/ctor test files/headers.

Reply:
I think this is a quite different scenario from existing constructor tests. Existing constructor tests test constructors in isolation. Here we have an interaction between three features: constructors, inline definitions of constructors, and inline functions. I think putting these tests (effectively integration tests) into separate files is the right choice.

Reply:
I agree that 'members' is not a good name for these tests but I failed to find a more suitable name and I'm happy to change it if a better one comes to mind. What these tests to is check that IR is generated for inline functions when called from constructors/members. I'd like to keep them separate from the rest of the tests on constructors and members for the reason @gribozavr mentioned above.

Reply:
> I think this is a quite different scenario from existing constructor tests. Existing constructor tests test constructors in isolation. Here we have an interaction between three features: constructors, inline definitions of constructors, and inline functions. I think putting these tests (effectively integration tests) into separate files is the right choice.

Sorry if I was unclear, I'm not suggesting that we merge these with the existing constructor tests. I'm saying that there are about half-a-dozen header files in this patch that all essentially test "a constructor that calls an inline function." I think all (or at least some) of these headers could be merged into one header. Then instead of having ~six headers that all test "a constructor that calls an inline function" via six different classes all called "Incrementor" we would have "inline-function-in-ctor.h" which would have classes called "CallsInlineFunction", "CallFunctionFromNestedStruct", etc. 

Either way, I think we should still probably put these tests in `Cxx/class`, as that is where the existing ctor/member tests are. At the very least, `Cxx/members` should be renamed to `Cxx/member` (no "s") to be in line with the rest of the C++ interop tests.

Reply:
I moved the tests into Cxx/class/inline-functions-through-member, as to keep them separate from future tests on members.

As for the headers - I am strongly in favor of keeping tests as isolated as possible.
Putting everything in one header will reduce readability when one looks at a single test (eg. me 6 months from now). And the similarity between the content of the header files doesn't make things better - no two headers call the `increment` function from the same place, so we'd need a constructor that calls the function, a method that calls the function, a nested struct with a constructor that calls the function etc. - the sum of all these is less readable than having them in separate headers.

Reply:
> I moved the tests into Cxx/class/inline-functions-through-member, as to keep them separate from future tests on members.

Fantastic. I really like using a subdirectory here.

> As for the headers - I am strongly in favor of keeping tests as isolated as possible.
Putting everything in one header will reduce readability when one looks at a single test (eg. me 6 months from now). And the similarity between the content of the header files doesn't make things better - no two headers call the increment function from the same place, so we'd need a constructor that calls the function, a method that calls the function, a nested struct with a constructor that calls the function etc. - the sum of all these is less readable than having them in separate headers.

Fair enough.